fetcher/fetcher_test.go

```diff
@@ -150,12 +149,18 @@ func TestFetchSameKeysSlow(t *testing.T) {
 			stateKeys.Add(strconv.Itoa(k), state.Read)
 		}
 		txID := ids.GenerateTestID()
-		if i%2 == 0 {
-			time.Sleep(1 * time.Second)
-		}
-		require.NoError(f.Fetch(ctx, txID, stateKeys))
-		go func(txID ids.ID) {
+
+		delay := make(chan struct{})
+		go func(txID ids.ID, i int) {
 			defer wg.Done()
+			if i%2 == 0 {
+				// do some work to minimic timeout
+				for i := 0; i < 10000000; i++ {
+				}
+				close(delay)
```

Comment:
logic is cleaner if you just always close this after the `if` statement (because you always create it)..then you don't need to do the `continue` below.